hathor/manager.py

```diff
@@ -120,6 +121,13 @@ def __init__(self, reactor: IReactorCore, peer_id: Optional[PeerId] = None, netw
 
         self.cpu = cpu
 
+        # XXX: first checkpoint must be genesis (height=0)
+        self.checkpoints: List[Checkpoint] = checkpoints or []
+        self.checkpoints_ready: List[bool] = [False] * len(self.checkpoints)
+        if not self.checkpoints or self.checkpoints[0].height > 0:
+            self.checkpoints.insert(0, Checkpoint(0, settings.GENESIS_BLOCK_HASH))
+            self.checkpoints_ready.insert(0, True)
```

Comment:
I feel we should either remove `checkpoints_ready` or add:

```python
if ...:
    :
    :
else:
    self.checkpoints_ready[0] = True
```

Reply:
I'd rather add the `else`. Otherwise the same goes `self.checkpoints`, not being used.

Reply:
Ok then! Even though I don't feel this information will be stored the manager, we can keep it here and refactor in the next PR.

Reply:
Done.